bindings/web5_uniffi_wrapper/src/credentials/verifiable_credential_1_1.rs

```diff
@@ -1,35 +1,110 @@
+pub struct VerifiableCredential(pub Arc<RwLock<InnerVerifiableCredential>>);
 
 impl VerifiableCredential {
-    pub fn new(verifiable_credential: InnerVerifiableCredential) -> Self {
-        Self(verifiable_credential)
+    pub fn new(verifiable_credential: data::VerifiableCredential) -> Result<Self> {
+        let inner_verifiable_credential = verifiable_credential.to_inner()?;
+
+        Ok(Self(Arc::new(RwLock::new(inner_verifiable_credential))))
     }
 
     pub fn verify(vcjwt: &str) -> Result<Self> {
-        let vc = InnerVerifiableCredential::verify(vcjwt).map_err(|e| Arc::new(e.into()))?;
-        Ok(Self(vc))
+        let inner_verifiable_credential =
+            InnerVerifiableCredential::verify(vcjwt).map_err(|e| Arc::new(e.into()))?;
+
+        Ok(Self(Arc::new(RwLock::new(inner_verifiable_credential))))
     }
 
     pub fn verify_with_verifier(vcjwt: &str, verifier: Arc<dyn Verifier>) -> Result<Self> {
-        let vc = InnerVerifiableCredential::verify_with_verifier(vcjwt, verifier.to_inner())
-            .map_err(|e| Arc::new(e.into()))?;
-        Ok(Self(vc))
+        let inner_verifiable_credential =
+            InnerVerifiableCredential::verify_with_verifier(vcjwt, verifier.to_inner())
+                .map_err(|e| Arc::new(e.into()))?;
+
+        Ok(Self(Arc::new(RwLock::new(inner_verifiable_credential))))
+    }
+
+    pub fn sign(&self, bearer_did: Arc<BearerDid>) -> Result<String> {
+        let inner_verifiable_credential = self
+            .0
+            .read()
+            .map_err(|e| RustCoreError::from_poison_error(e, "RwLockReadError"))?;
+
+        inner_verifiable_credential
+            .sign(&bearer_did.0)
+            .map_err(|e| Arc::new(e.into()))
     }
 
-    pub fn sign(&self, signer: Arc<dyn Signer>) -> Result<String> {
-        self.0
-            .sign(signer.to_inner())
+    pub fn sign_with_signer(&self, key_id: &str, signer: Arc<dyn Signer>) -> Result<String> {
+        let inner_verifiable_credential = self
+            .0
+            .read()
+            .map_err(|e| RustCoreError::from_poison_error(e, "RwLockReadError"))?;
+
+        inner_verifiable_credential
+            .sign_with_signer(key_id, signer.to_inner())
             .map_err(|e| Arc::new(e.into()))
     }
 
-    pub fn get_data(&self) -> InnerVerifiableCredential {
-        self.0.clone()
+    pub fn get_data(&self) -> Result<data::VerifiableCredential> {
+        let inner_verifiable_credential = self
+            .0
+            .read()
+            .map_err(|e| RustCoreError::from_poison_error(e, "RwLockReadError"))?;
+
+        data::VerifiableCredential::from_inner(inner_verifiable_credential.clone())
+    }
+}
+
+pub mod data {
+    use super::*;
+    use std::time::SystemTime;
+
+    #[derive(Clone)]
+    pub struct VerifiableCredential {
+        pub context: Vec<String>,
+        pub id: String,
+        pub r#type: Vec<String>,
+        pub issuer: String, // JSON serialized
```

Comment:
This could cause confusion on if its an issuer string "did:dht:123" or an json serialized json string.

Should we rename `json_serialized_issuer` ?

I think sticking with the official name has value too though

Reply:
Oh yeah wow I really appreciate this idea... I think yes we should follow that practice everywhere wherein we're json serializing over the FFI (including over in tbdex, going to open a PR for that now...)

Reply:
nice, yeah I really like this idea https://github.com/TBD54566975/tbdex-rs/pull/63